# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/RetainedDisplayListBuilder.cpp
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 layout/painting/RetainedDisplayListBuilder.cpp
--- a/layout/painting/RetainedDisplayListBuilder.cpp	Fri Mar 22 18:28:31 2019 +0000
+++ b/layout/painting/RetainedDisplayListBuilder.cpp	Fri Mar 22 18:28:42 2019 +0000
@@ -364,7 +364,7 @@
           if (mBuilder->MergeDisplayLists(
                   aNewItem->GetChildren(), oldItem->GetChildren(),
                   destItem->GetChildren(), containerASRForChildren, aNewItem)) {
-            destItem->InvalidateCachedChildInfo();
+            destItem->InvalidateCachedChildInfo(mBuilder->Builder());
             mResultIsModified = true;
           }
           UpdateASR(destItem, containerASRForChildren);
@@ -504,6 +504,7 @@
       Span<const MergedListIndex> aDirectPredecessors,
       const Maybe<MergedListIndex>& aExtraDirectPredecessor) {
     UpdateContainerASR(aItem);
+    aItem->NotifyUsed(mBuilder->Builder());
 
 #ifdef MOZ_DIAGNOSTIC_ASSERT_ENABLED
     nsIFrame::DisplayItemArray* items =
@@ -542,7 +543,7 @@
         if (mBuilder->MergeDisplayLists(&empty, item->GetChildren(),
                                         item->GetChildren(),
                                         containerASRForChildren, item)) {
-          item->InvalidateCachedChildInfo();
+          item->InvalidateCachedChildInfo(mBuilder->Builder());
           mResultIsModified = true;
         }
         UpdateASR(item, containerASRForChildren);